File: apps/react-native/src/test/fixtures.ts
import type { BabyInput, NewRecordInput } from '../domain/types';

export const babyInputFixture = (): BabyInput => ({
  name: '安安',
  birthDate: '2026-01-15',
  sex: 'female',
  avatarPath: 'file:///avatars/anan.jpg',
});

export const momentInputFixture = (
  overrides: Partial<NewRecordInput> = {},
): NewRecordInput => ({
  type: 'moment',
  occurredAt: '2026-08-01T09:30:00.000Z',
  note: '第一次看向镜头',
  details: null,
  attachments: [
    {
      id: 'attachment-image-1',
      mediaType: 'image',
      filePath: 'file:///media/first-look.jpg',
      thumbnailPath: 'file:///media/first-look-thumb.jpg',
    },
  ],
  ...overrides,
});

export const recordVariantFixtures = (): NewRecordInput[] => [
  momentInputFixture(),
  {
    type: 'growth',
    occurredAt: '2026-08-02T09:30:00.000Z',
    note: null,
    details: { heightCm: 66.2, weightKg: 7.4, headCm: null },
    attachments: [],
  },
  {
    type: 'activity',
    occurredAt: '2026-08-03T09:30:00.000Z',
    note: '午后小睡',
    details: { activityType: 'sleep', amount: null, durationMinutes: 45 },
    attachments: [],
  },
  {
    type: 'milestone',
    occurredAt: '2026-08-04T09:30:00.000Z',
    note: null,
    details: { title: '会翻身', presetKey: 'roll-over' },
    attachments: [],
  },
];

File: apps/flutter/test/support/fixtures.dart
import 'package:baby_growth_timeline/domain/models/attachment.dart';
import 'package:baby_growth_timeline/domain/models/baby.dart';
import 'package:baby_growth_timeline/domain/models/record_draft.dart';
import 'package:baby_growth_timeline/domain/models/timeline_record.dart';

final fixtureOccurredAt = DateTime.parse('2026-07-31T22:15:00+08:00');

BabyDraft babyDraftFixture({String name = '宝宝'}) => BabyDraft(
  name: name,
  birthDate: '2025-06-12',
  sex: '女',
  avatarPath: '/avatars/baby.jpg',
);

NewRecordInput momentInputFixture({
  String? note = '第一次微笑',
  DateTime? occurredAt,
  List<NewAttachmentInput>? attachments,
}) => NewRecordInput(
  type: RecordType.moment,
  occurredAt: occurredAt ?? fixtureOccurredAt,
  note: note,
  attachments:
      attachments ??
      const [
        NewAttachmentInput(
          id: 'attachment-moment',
          mediaType: MediaType.image,
          filePath: '/media/smile.jpg',
          thumbnailPath: '/media/smile-thumb.jpg',
        ),
      ],
);

NewRecordInput growthInputFixture({DateTime? occurredAt}) => NewRecordInput(
  type: RecordType.growth,
  occurredAt: occurredAt ?? fixtureOccurredAt,
  note: '满六个月',
  details: const RecordDetails.growth(
    heightCm: 68.5,
    weightKg: 7.4,
    headCm: 42.1,
  ),
);

NewRecordInput activityInputFixture({DateTime? occurredAt}) => NewRecordInput(
  type: RecordType.activity,
  occurredAt: occurredAt ?? fixtureOccurredAt,
  note: '午觉',
  details: const RecordDetails.activity(
    activityType: ActivityType.sleep,
    amount: 120,
    durationMinutes: 95,
  ),
);

NewRecordInput milestoneInputFixture({DateTime? occurredAt}) => NewRecordInput(
  type: RecordType.milestone,
  occurredAt: occurredAt ?? fixtureOccurredAt,
  note: '值得纪念',
  details: const RecordDetails.milestone(
    title: '第一次独站',
    presetKey: 'stand-alone',
  ),
);
